ballerina/tests/to_xml_test.bal

```diff
@@ -0,0 +1,852 @@
+    xml result = check toXml(data);
+    test:assertEquals(result.toString(), expected, msg = "testMapFloatArrayToXml result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapJsonArrayToXml1() returns error? {
+    map<json[]> data = {customer: [{name: "Asha", age: 10}, {name: "Kalai", age: 12}]};
+    string expected = "<root>" +
+                        "<customer>" +
+                            "<name>Asha</name>" +
+                            "<age>10</age>" +
+                        "</customer>" +
+                        "<customer>" +
+                            "<name>Kalai</name>" +
+                            "<age>12</age>" +
+                        "</customer>" +
+                    "</root>";
+    xml result = check toXml(data);
+    test:assertEquals(result.toString(), expected, msg = "testMapJsonToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testMapXmlArrayToXml1() returns error? {
+    map<xml[]> data = {
+        customers:
+        [
+            xml `<customer><name>Asha</name><age>10</age></customer>`,
+            xml `<customer><name>Kalai</name><age>13</age></customer>`
+        ],
+        employees:
+        [
+            xml `<employee><name>Asha</name><age>10</age></employee>`,
+            xml `<employee><name>Kalai</name><age>13</age></employee>`
+        ]
+    };
+    string expected = "<root>" +
+                        "<customers>" +
+                            "<customer>" +
+                                "<name>Asha</name>" +
+                                "<age>10</age>" +
+                            "</customer>" +
+                            "<customer>" +
+                                "<name>Kalai</name>" +
+                                "<age>13</age>" +
+                            "</customer>" +
+                        "</customers>" +
+                        "<employees>" +
+                            "<employee>" +
+                                "<name>Asha</name>" +
+                                "<age>10</age>" +
+                            "</employee>" +
+                            "<employee>" +
+                                "<name>Kalai</name>" +
+                                "<age>13</age>" +
+                            "</employee>" +
+                        "</employees>" +
+                    "</root>";
+    xml result = check toXml(data);
+    test:assertEquals(result.toString(), expected, msg = "testMapXmlArrayToXml1 result incorrect");
+}
+
+@test:Config {
+    groups: ["toXml"]
+}
+isolated function testRecordArrayToXml1() returns error? {
+    Customer[] customers = [{ns\:name: "Asha", age: 10}, {ns\:name: "Kalai", age: 10}];
+    map<Customer[]> data = {customers: customers};
+    string expected = "<root>" +
+                        "<customers>" +
+                            "<ns:Customers xmlns:ns=\"http://sdf.com\" ns:employeeName=\"Asha\">" +
```

Comment:
Why do we need two elements here?

Reply:
Or do we need <root> element here. because we have one element called <customers>

Reply:
The second sample is correct.  Fixed in [e9ed663](https://github.com/ballerina-platform/module-ballerina-xmldata/pull/431/commits/e9ed663e63027a80cd23907bc746b35aae654804)